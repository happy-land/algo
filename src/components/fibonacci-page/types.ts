export interface FiboStep {
  numbers: number[];
  index?: number;
} 