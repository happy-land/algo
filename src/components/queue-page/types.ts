import { ElementStates } from "../../types/element-states";

export type TElement<T> = {
  value: T;
  state?: ElementStates;
};
