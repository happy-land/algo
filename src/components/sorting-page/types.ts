import { ElementStates } from "../../types/element-states";

export type TSortMode = 'selection' | 'bubble';

export type TElement<T> = {
  value: T;
  state?: ElementStates;
};

export interface SortingStep<T> {
  elements: TElement<T>[]; // элементы
}