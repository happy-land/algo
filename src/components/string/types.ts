import { ElementStates } from "../../types/element-states";

export interface LettersStep {
  letters: string[];
  index?: number;
  state?: ElementStates;
}