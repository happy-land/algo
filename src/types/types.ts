import { ElementStates } from "./element-states";

export type TCircle = {
  letter?: string;
  index?: number;
  state?: ElementStates;
}