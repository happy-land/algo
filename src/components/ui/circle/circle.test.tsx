import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";
import { Circle } from "./circle";

describe("CIRCLE UI COMPONENT TEST", () => {
  test("Circle without letter", () => {
    const circleWithoutLetter = renderer.create(<Circle />);
    expect(circleWithoutLetter).toMatchSnapshot();
  });

  test("Circle with letter", () => {
    const circleWithLetter = renderer.create(<Circle letter="q" />);
    expect(circleWithLetter).toMatchSnapshot();
  });

  test("Circle with head", () => {
    const container = render(<Circle head="asd" />);
    const circleWithHead = container.getByText("asd");
    expect(circleWithHead.textContent).toEqual("asd");
    expect(container).toMatchSnapshot();
  });

  test("Circle with react element in head", () => {
    const circleWithReactEl = renderer.create(<Circle head={<Circle />} />);
    expect(circleWithReactEl).toMatchSnapshot();
  });

  test("Circle with tail", () => {
    const circleWithTail = renderer.create(<Circle tail="tail" />);
    expect(circleWithTail).toMatchSnapshot();
  });

  test("Circle with react element in tail", () => {
    const circleWithReactEl = renderer.create(<Circle tail={<Circle />} />);
    expect(circleWithReactEl).toMatchSnapshot();
  });

  test("Circle with index", () => {
    const circleWithIndex = renderer.create(<Circle index={4} />);
    expect(circleWithIndex).toMatchSnapshot();
  });

  test("Circle with isSmall props", () => {
    const circleSmall = renderer.create(<Circle isSmall />);
    expect(circleSmall).toMatchSnapshot();
  });

  test("Circle with Default state", () => {
    const circleDefault = renderer.create(<Circle state={ElementStates.Default} />);
    expect(circleDefault).toMatchSnapshot();
  });

  test("Circle with Changing state", () => {
    const circleChanging = renderer.create(<Circle state={ElementStates.Changing} />);
    expect(circleChanging).toMatchSnapshot();
  });

  test("Circle with Modified state", () => {
    const circleModified = renderer.create(<Circle state={ElementStates.Modified} />);
    expect(circleModified).toMatchSnapshot();
  });
});
