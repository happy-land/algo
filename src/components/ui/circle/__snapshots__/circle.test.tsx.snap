// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CIRCLE UI COMPONENT TEST Circle with Changing state 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with Default state 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with Modified state 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with head 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-cy="circle-wrapper"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head string"
          data-testid="circle_head"
        >
          asd
        </div>
        <div
          class="circle   default"
          data-cy="circle"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="circle_letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-testid="circle_tail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-cy="circle-wrapper"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head string"
        data-testid="circle_head"
      >
        asd
      </div>
      <div
        class="circle   default"
        data-cy="circle"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="circle_letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circle_tail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CIRCLE UI COMPONENT TEST Circle with index 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    4
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with isSmall props 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle  small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with letter 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    >
      q
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with react element in head 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  >
    <div
      className="content "
      data-cy="circle-wrapper"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circle_head"
      />
      <div
        className="circle   default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle_letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circle_tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with react element in tail 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  >
    <div
      className="content "
      data-cy="circle-wrapper"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circle_head"
      />
      <div
        className="circle   default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle_letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circle_tail"
      />
    </div>
  </div>
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle with tail 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circle_tail"
  >
    tail
  </div>
</div>
`;

exports[`CIRCLE UI COMPONENT TEST Circle without letter 1`] = `
<div
  className="content "
  data-cy="circle-wrapper"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle_head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle_tail"
  />
</div>
`;
